sympy/vector/coordsysrect.py

```diff
@@ -98,6 +119,27 @@ def __new__(cls, name, location=None, rotation_matrix=None,
             location = Vector.zero
             origin = Point(name + '.origin')
 
+        if transformation is None:
+            transformation = Tuple(rotation_matrix, location)
+
+        if isinstance(transformation, Tuple):
```

Comment:
The main problem is that when calling the superclass in the constructor, we need a fixed argument order. The same argument order must return the same object if called on the constructor. This requirement is pretty restrictive on the way we may write the constructor, so we need to choose a precise way to pack the information without losing speed.